### DISCUSSION TITLE
KeyWord search in Faceted Search

### DISCUSSION THREAD

**Discussion by Cameros-agency** [PrestaShop/PrestaShop#40209] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

I didn't find any option to enable a filter by keyword in the faceted_search module. I think it would be pretty nice when a category page still has hundreds of products after filtering by a feature value. 

### Describe the solution you'd like

It would be nice to have a keyword filter in the layered filter configuration. 

![image](https://user-images.githubusercontent.com/23308427/191243700-1b256224-d00d-4dd8-9358-775168bb87b5.png)

### Additional context

When your category page has lots of products  (in my case a vehicle spare parts shop) the features and features value filter system are pretty efficient. 
However, we still encounter some cases where the user has hundreds of products after using filters so a search by keyword would be a plus to help the final user to find the product he needs.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by paulnoelcholot** [comment]
Hello @CIMEOS 

Can you suggest me a design for the location of the keyword filter?
Can you specify what kind of keyword you wanna use? :pray: 

Waiting for your feedback.

Thanks

**Comment by Cameros-agency** [comment]
Hi @paulnoelcholot,
Thank you for your quick response!

> Can you suggest me a design for the location of the keyword filter?

On the frontend, I guess the design could look like a simple input field inside the left column. (if the faceted search is in the left column). 
I attach a screenshot with a potential location but the best would be to be able to change the location using the current drag and drop system.
![Prestashop_keyword_filter](https://user-images.githubusercontent.com/6168624/191205585-1bbaf478-eac8-4797-b77c-8ebffba0ee40.jpg)


> Can you specify what kind of keyword you wanna use? 🙏

The best would be to search in the product name, the product description and short description, and the product reference.
So I guess the request should be named search by a search string instead of keyword.

Hope this will help!

**Comment by hibatallahAouadni** [comment]
Hello @CIMEOS 

Thank you for your suggestion. The Maintainer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by Hlavtox** [comment]
@CIMEOS I think a better solution would be to make facetedsearch work on all pages including search page. (https://github.com/PrestaShop/PrestaShop/issues/15137)

This is a number one thing my clients want.

**Comment by Cameros-agency** [comment]
> @CIMEOS I think a better solution would be to make facetedsearch work on all pages including search page. (#15137)

Hi @Hlavtox,
In the long term, it would indeed be a better solution! But if you think about it, some users maybe don't want to put the faceted search on all pages so it could be a plus.
